pkg/kubectl/cmd/apply.go

```diff
@@ -345,12 +348,15 @@ func RunApply(f cmdutil.Factory, cmd *cobra.Command, out, errOut io.Writer, opti
 
 			if string(patchBytes) == "{}" {
 				count++
-				cmdutil.PrintSuccess(shortOutput, out, info.Object, false, "unchanged")
+
+				if !printObject {
+					cmdutil.PrintSuccess(shortOutput, out, info.Object, false, "unchanged")
+				}
 				return nil
```

Comment:
This will return here when it's unchanged, which means that `kubectl apply -f test.yaml -o yaml` (example in #58836) will print nothing. It should still print object in this case. Therefore, we need to move `return nil` into the if-block.

Reply:
I was hoping not to do that, as this would break some stuff we have around assuming unchanged objects not being printed, but I guess what you suggested is the correct behavior. I will change it